### PR TITLE
netmiko poxy import fixes

### LINKED ISSUE #62405: [BUG] netmiko proxy exception depending on version installed
**Description**
If a user attempts to use the Netmiko proxy minion with the version of Netmiko Python library higher than 3.4.0 then an exception is caused and the proxy minion is unable to load.

```
[DEBUG   ] Failed to import proxy netmiko_px:
Traceback (most recent call last):
  File "salt/loader/lazy.py", line 768, in _load_module
    mod = self.run(spec.loader.load_module)
  File "salt/loader/lazy.py", line 1228, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "salt/loader/lazy.py", line 1243, in _run_as
    return _func_or_method(*args, **kwargs)
  File "<frozen importlib._bootstrap_external>", line 529, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 1029, in load_module
  File "<frozen importlib._bootstrap_external>", line 854, in load_module
  File "<frozen importlib._bootstrap>", line 274, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/saltstack/salt/run/salt/proxy/netmiko_px.py", line 195, in <module>
    from netmiko.ssh_exception import NetMikoTimeoutException
ModuleNotFoundError: No module named 'netmiko.ssh_exception'
```

**Setup**
1/ Install Netmiko version higher than 3.4.0
2. Setup the pillar files for a simple Netmiko proxy minion:
```
proxy:
  proxytype: netmiko
  device_type: linux
  host: 127.0.0.1
  username: saltdev
  password: 'saltdev'
  always_alive: False
```
3. Start the Salt proxy minion using the above mentioned minion id.

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Due to changes in the Netmiko library for the exception paths, need to check the version of Netmiko python library and then import the exceptions from different locations depending on the result.

### What issues does this PR fix or reference?
Fixes: #62405 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
Change in Netmiko mentioned here: https://github.com/ktbyers/netmiko/releases/tag/v4.0.0

**Review by whytewolf — DISMISSED** [review]

**Review by dmurphy18 — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]
changelog?

File: salt/proxy/netmiko_px.py [hunk 1]
```diff
@@ -187,13 +187,19 @@
...
 
 try:
     from netmiko import ConnectHandler
-    from netmiko.ssh_exception import (
-        NetMikoAuthenticationException,
-        NetMikoTimeoutException,
-    )
+    from netmiko import __version__ as found_netmiko_version
+
+    if _LooseVersion(found_netmiko_version) <= _LooseVersion("3.4.0"):
```
Comment (s0undt3ch) [inline salt/proxy/netmiko_px.py:196]: I'd do a simple `try/catch ImportError`, it's faster and doesn't require version comparisons.
Try to import from the newer module, fallback to the old.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Tests are passing and includes changelog.
Merging.

**Comment by s0undt3ch** [comment]
Dam, @garethgreenaway looks like you need to rebase.

**Pull request merged**